pkg/manifests/manifests.go

```diff
@@ -883,7 +884,23 @@ func (f *Factory) updateNodeExporterArgs(args []string) []string {
 		args = setArg(args, "--no-collector.ksmd", "")
 	}
 
-	return args
+	if f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Enabled {
+		args = setArg(args, "--collector.systemd", "")
+		units := f.config.ClusterMonitoringConfiguration.NodeExporterConfig.Collectors.Systemd.Units
+		for idx, unit := range units {
+			_, err := regexp.Compile(unit)
+			if err != nil {
+				return nil, fmt.Errorf("invalid regexp for systemd unit: %s", unit)
+			}
+			units[idx] = fmt.Sprintf("(%s)", unit)
```

Comment:
do we need the enclosing brackets?

Reply:
After a second thought, any pattern that needs brackets to limit cannot pass the regex check just before this line. No bracket will be added.